### PR TITLE
Add an integration test to test Cfn API with localstack.

### PR DESCRIPTION
**Pull request by theghostwhoforks**
Enables an integration test suite to verify CFn API interaction with localstack.

The test creates a S3 bucket, *PUT*s a JSON template file that can provision a S3 bucket in localstack. Individual tests then creates a stack based on the uploaded template JSON file and verifies Cloudformation API interaction.

There is no teardown implemented yet, as the lifecycle of the docker image is scoped per test class.

### CODE REVIEW

**Comment by bpar476** [comment]
I tried to run this locally and got this error. However, I also got it for all the other ITs

```
[ERROR] com.atlassian.migration.datacenter.core.aws.CfnApiIT  Time elapsed: 24.696 s  <<< ERROR!
java.lang.reflect.InaccessibleObjectException: Unable to make static void com.atlassian.migration.datacenter.core.aws.CfnApiIT.beforeAll() throws java.io.IOException accessible: module org.codehaus.stax2 does not "opens com.atlassian.migration.datacenter.core.aws" to unnamed module @1caa0244
```

**Comment by theghostwhoforks** [comment]
> java.lang.reflect.InaccessibleObjectException: Unable to make static void com.atlassian.migration.datacenter.core.aws.CfnApiIT.beforeAll() throws java.io.IOException accessible: module org.codehaus.stax2 does not "opens com.atlassian.migration.datacenter.core.aws" to unnamed module @1caa0244

oh wow.  Let me clean and run again

**Comment by theghostwhoforks** [comment]
> I tried to run this locally and got this error. However, I also got it for all the other ITs

Did you run this from IDEA or the CLI?

**Comment by bpar476** [comment]
> I tried to run this locally and got this error. However, I also got it for all the other ITs
> 
> ```
> [ERROR] com.atlassian.migration.datacenter.core.aws.CfnApiIT  Time elapsed: 24.696 s  <<< ERROR!
> java.lang.reflect.InaccessibleObjectException: Unable to make static void com.atlassian.migration.datacenter.core.aws.CfnApiIT.beforeAll() throws java.io.IOException accessible: module org.codehaus.stax2 does not "opens com.atlassian.migration.datacenter.core.aws" to unnamed module @1caa0244
> ```

Hmm, this happened with unit tests as well. I've just done a mvn clean and will try again.

File: src/test/java/com/atlassian/migration/datacenter/core/aws/CfnApiIT.java [hunk 1]
```diff
@@ -26,38 +45,52 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-
-/**
- * Test Integration test. No mocks involved
- */
+@Tag("integration")
+@ExtendWith({LocalstackDockerExtension.class})
+@LocalstackDockerProperties(services = {"cloudformation", "s3"}, imageTag = "0.10.6")
```
Comment (nanux) [inline src/test/java/com/atlassian/migration/datacenter/core/aws/CfnApiIT.java]: I would recommend making this `0.10.8`, new localstack works better with the newer image. I will merge the new version shortly (together with the integration test in github actions).

**Comment by bpar476** [comment]
Okay, got around the first issue but now the CredentialsProvider isn't resolving credentials. I think the approach in the other IT's is to implement a stub credentials provider which just returns empty strings for access key ID and secret access key

File: src/test/java/com/atlassian/migration/datacenter/core/aws/CfnApiIT.java (hunk 1 above)
Comment (theghostwhoforks) [reply src/test/java/com/atlassian/migration/datacenter/core/aws/CfnApiIT.java]: Done

**Comment by theghostwhoforks** [comment]
> Okay, got around the first issue but now the CredentialsProvider isn't resolving credentials. I think the approach in the other IT's is to implement a stub credentials provider which just returns empty strings for access key ID and secret access key

Sorted this out

**Pull request merged**